src/main/kotlin/com/goodchoice/domain/user/service/EmployeeSelfService.kt

```diff
@@ -0,0 +1,39 @@
+package com.goodchoice.domain.user.service
+
+import com.goodchoice.domain.auth.service.AuthService
+import com.goodchoice.domain.user.model.EmployeeInvitationSelfView
+import com.goodchoice.domain.user.model.EmployeeRegistrationRequest
+import com.goodchoice.domain.user.model.toSelfView
+import com.goodchoice.domain.user.persistence.EmployeeRepository
+import org.springframework.transaction.annotation.Transactional
+
+interface EmployeeSelfService {
+
+    fun getInvitationByToken(token: String): EmployeeInvitationSelfView
+
+    fun acceptInvitation(request: EmployeeRegistrationRequest)
+}
+
+class EmployeeSelfServiceImpl(
+    private val employeeRepo: EmployeeRepository,
+    private val invitationService: EmployeeInvitationService,
+    private val authService: AuthService,
+) : EmployeeSelfService {
+
+    @Transactional(readOnly = true)
```

Comment:
What does readOnly parameter?

Reply:
According to `Transactional` [docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Transactional.html):
A boolean flag that can be set to true if the transaction is effectively read-only, allowing for corresponding optimizations at runtime.